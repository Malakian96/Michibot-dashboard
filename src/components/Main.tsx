/* eslint-disable @typescript-eslint/no-explicit-any */
import GuildSelector from "./GuildSelector";
import ChannelSelector from "./ChannelSelector";
import SoundBoard from "./Soundboard";

const Main = ({
  guilds,
  selectedGuild,
  setSelectedGuild,
  channels,
  setChannels,
  selectedChannel,
  setSelectedChannel,
  handlePlaySound,
  isButtonDisabled,
  audios,
}: any) => {
  return (
    <div className="mt-6">
      <div className="flex justify-around flex-wrap gap-5">
        <GuildSelector
          guilds={guilds}
          selectedGuild={selectedGuild}
          setSelectedGuild={setSelectedGuild}
          setChannels={setChannels}
        />
        <ChannelSelector
          channels={channels}
          selectedChannel={selectedChannel}
          setSelectedChannel={setSelectedChannel}
        />
      </div>
      <SoundBoard
        audios={audios}
        handlePlaySound={handlePlaySound}
        isButtonDisabled={isButtonDisabled}
      />
    </div>
  );
};

export default Main;
