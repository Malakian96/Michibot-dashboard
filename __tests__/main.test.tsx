import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Main from '../src/components/Main';

const mockGuilds = [
  {
    guildId: '1',
    guildName: 'Guild 1',
    voiceChannels: [
      { id: '1', name: 'Channel 1' },
      { id: '2', name: 'Channel 2' },
    ],
  },
  {
    guildId: '2',
    guildName: 'Guild 2',
    voiceChannels: [
      { id: '3', name: 'Channel 3' },
      { id: '4', name: 'Channel 4' },
    ],
  },
];

const mockAudios = ['Audio 1', 'Audio 2'];

describe('Main', () => {
  it('renders correctly', () => {
    const { getByText } = render(
      <Main
        guilds={mockGuilds}
        selectedGuild=""
        setSelectedGuild={() => {}}
        channels={[]}
        setChannels={() => {}}
        selectedChannel=""
        setSelectedChannel={() => {}}
        handlePlaySound={() => {}}
        isButtonDisabled={false}
        audios={mockAudios}
      />,
    );
    expect(getByText('Select a server')).toBeInTheDocument();
    expect(getByText('Select a voice channel')).toBeInTheDocument();
  });

  it('calls setSelectedGuild and setChannels on guild change', () => {
    const setSelectedGuild = jest.fn();
    const setChannels = jest.fn();

    const { getByRole } = render(
      <Main
        guilds={mockGuilds}
        selectedGuild=""
        setSelectedGuild={setSelectedGuild}
        channels={[]}
        setChannels={setChannels}
        selectedChannel=""
        setSelectedChannel={() => {}}
        handlePlaySound={() => {}}
        isButtonDisabled={false}
        audios={mockAudios}
      />,
    );

    fireEvent.change(getByRole('combobox', { name: /Select a server/i }), {
      target: { value: '1' },
    });

    expect(setSelectedGuild).toHaveBeenCalledWith('1');
    expect(setChannels).toHaveBeenCalledWith(mockGuilds[0].voiceChannels);
  });

  it('calls handlePlaySound on button click', () => {
    const handlePlaySound = jest.fn();

    const { getByTestId } = render(
      <Main
        guilds={mockGuilds}
        selectedGuild="1"
        setSelectedGuild={() => {}}
        channels={mockGuilds[0].voiceChannels}
        setChannels={() => {}}
        selectedChannel="1"
        setSelectedChannel={() => {}}
        handlePlaySound={handlePlaySound}
        isButtonDisabled={false}
        audios={mockAudios}
      />,
    );

    fireEvent.click(getByTestId('sound-buttonAudio 1'));

    expect(handlePlaySound).toHaveBeenCalledWith('Audio 1');
  });
});

describe('MainSnapshot', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <Main
        guilds={mockGuilds}
        selectedGuild=""
        setSelectedGuild={() => {}}
        channels={[]}
        setChannels={() => {}}
        selectedChannel=""
        setSelectedChannel={() => {}}
        handlePlaySound={() => {}}
        isButtonDisabled={false}
        audios={mockAudios}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
