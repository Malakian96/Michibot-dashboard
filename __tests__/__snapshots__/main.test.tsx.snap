// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSnapshot matches snapshot 1`] = `
<div>
  <div
    class="mt-6"
  >
    <div
      class="flex justify-around flex-wrap gap-5"
    >
      <select
        aria-label="Select a server"
        class="bg-discord-dark-light text-white px-4 py-2 rounded-md mb-4"
      >
        <option
          value=""
        >
          Select a server
        </option>
        <option
          value="1"
        >
          Guild 1
        </option>
        <option
          value="2"
        >
          Guild 2
        </option>
      </select>
      <select
        class="bg-discord-dark-light text-white px-4 py-2 rounded-md mb-4"
        disabled=""
      >
        <option
          value=""
        >
          Select a voice channel
        </option>
      </select>
    </div>
    <div
      class="mt-4 grid grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4"
    >
      <div
        class="flex flex-col items-center"
      >
        <button
          class="w-32 h-32 rounded-md overflow-hidden hover:scale-105 transition-transform"
          data-testid="sound-buttonAudio 1"
          style="background-image: url(https://media.giphy.com/media/JIX9t2j0ZTN9S/giphy.gif); background-size: cover; background-position: center;"
        />
        <span
          class="text-xs text-white mt-1"
        >
          Audio 1
        </span>
      </div>
      <div
        class="flex flex-col items-center"
      >
        <button
          class="w-32 h-32 rounded-md overflow-hidden hover:scale-105 transition-transform"
          data-testid="sound-buttonAudio 2"
          style="background-image: url(https://media.giphy.com/media/mlvseq9yvZhba/giphy.gif); background-size: cover; background-position: center;"
        />
        <span
          class="text-xs text-white mt-1"
        >
          Audio 2
        </span>
      </div>
    </div>
  </div>
</div>
`;
